### PR TITLE
Making sure dedxIh_StripOnlyFromGeneralTrack exists before calling numberOfMeasurements on it

### PR DESCRIPTION
**Pull request by tvami**

### CODE REVIEW

File: Analyzer/plugins/Analyzer.cc [hunk 1]
```diff
@@ -5892,6 +5893,7 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
                     false, 0, false, false, true, pixelCPE_, tTopo, generalTrack->px(), generalTrack->py(), generalTrack->pz(), generalTrack->charge());
 
       reco::DeDxData* dedxIh_StripOnlyFromGeneralTrack = dedxIh_StripOnly_TmpFromGeneralTrack.numberOfMeasurements() > 0 ? &dedxIh_StripOnly_TmpFromGeneralTrack : nullptr;
+      // TAV: Shouldnt we exit the loop in dedxIh_StripOnlyFromGeneralTrack is a nullptr? Caroline?
```
Comment (tvami) [inline Analyzer/plugins/Analyzer.cc:5896]: @carolinecollard  ?

File: Analyzer/plugins/Analyzer.cc [hunk 2]
```diff
@@ -5996,7 +5998,8 @@ void Analyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup)
     
       // nonL1PixHitsFromGeneralTrack now well computed above in the loop
 //      unsigned int numDeDxHitsFromGeneralTrack = dedxIh_noL1_TmpFromGeneralTrack.numberOfMeasurements() ;
-      unsigned int numDeDxHitsFromGeneralTrack = dedxIh_StripOnlyFromGeneralTrack->numberOfMeasurements() + nonL1PixHitsFromGeneralTrack;
```
Comment (tvami) [inline Analyzer/plugins/Analyzer.cc:5999]: This possibly called `numberOfMeasurements` on a `nullptr`

**Comment by carolinecollard** [comment]
Would you have the possibility to run of several samples which crash, to check if this problem was causing all the crashes ?

File: Analyzer/plugins/Analyzer.cc (hunk 1 above)
Comment (tvami) [reply Analyzer/plugins/Analyzer.cc:5896]: @carolinecollard but if this happens, do you agree we should have a continue?

**Pull request merged**